File: clients/java/src/main/java/spb_ai_champ/model/Player.java
package spb_ai_champ.model;

import spb_ai_champ.util.StreamUtil;

/**
 * TODO - Document
 */
public class Player {
    /**
     * TODO - Document
     */
    private int score;

    /**
     * TODO - Document
     */
    public int getScore() {
        return score;
    }

    /**
     * TODO - Document
     */
    public void setScore(int value) {
        this.score = value;
    }

    public Player(int score) {
        this.score = score;
    }

    /**
     * Read Player from input stream
     */
    public static Player readFrom(java.io.InputStream stream) throws java.io.IOException {
        int score;
        score = StreamUtil.readInt(stream);
        return new Player(score);
    }

    /**
     * Write Player to output stream
     */
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeInt(stream, score);
    }

    /**
     * Get string representation of Player
     */
    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder("Player { ");
        stringBuilder.append("score: ");
        stringBuilder.append(String.valueOf(score));
        stringBuilder.append(" }");
        return stringBuilder.toString();
    }
}

File: clients/kotlin/src/main/kotlin/spb_ai_champ/model/Player.kt
package spb_ai_champ.model

import spb_ai_champ.util.StreamUtil

/**
 * Player (game participant)
 */
class Player {
    /**
     * Team index
     */
    var teamIndex: Int
    /**
     * Current score points
     */
    var score: Int
    /**
     * Player's specialty
     */
    var specialty: spb_ai_champ.model.Specialty?

    constructor(teamIndex: Int, score: Int, specialty: spb_ai_champ.model.Specialty?) {
        this.teamIndex = teamIndex
        this.score = score
        this.specialty = specialty
    }

    /**
     * Write Player to output stream
     */
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, teamIndex)
        StreamUtil.writeInt(stream, score)
        val specialtyValue = specialty
        if (specialtyValue == null) {
            StreamUtil.writeBoolean(stream, false)
        } else {
            StreamUtil.writeBoolean(stream, true)
            StreamUtil.writeInt(stream, specialtyValue.tag)
        }
    }

    /**
     * Get string representation of Player
     */
    override fun toString(): String {
        var stringBuilder = StringBuilder("Player { ")
        stringBuilder.append("teamIndex: ")
        stringBuilder.append(teamIndex)
        stringBuilder.append(", ")
        stringBuilder.append("score: ")
        stringBuilder.append(score)
        stringBuilder.append(", ")
        stringBuilder.append("specialty: ")
        stringBuilder.append(specialty)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        /**
         * Read Player from input stream
         */
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Player {
            var teamIndex: Int
            teamIndex = StreamUtil.readInt(stream)
            var score: Int
            score = StreamUtil.readInt(stream)
            var specialty: spb_ai_champ.model.Specialty?
            if (StreamUtil.readBoolean(stream)) {
                specialty = spb_ai_champ.model.Specialty.readFrom(stream)
            } else {
                specialty = null
            }
            return Player(teamIndex, score, specialty)
        }
    }
}